src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -206,7 +207,7 @@ private void refreshSecretFromCloud(String arn, String versionStage) {
         String versionLabel = Utils.isEmpty(versionStage) ? LATEST_LABEL : versionStage;
         List<SecretConfiguration> configurations = getSecretConfiguration();
         boolean isSecretLabelConfigured = configurations.stream().anyMatch(
-                (secret) -> arn.contains(secret.getArn()) && secret.getLabels().contains(versionStage));
+                (secret) -> secret.getArn().equalsIgnoreCase(arn) && secret.getLabels().contains(versionStage));
```

Comment:
We want to know if a secret is configured with a partial arn or a friendly name as well. So, you'll need to use `contains`.

Reply:
Updated